flexmeasures/api/common/schemas/sensor_data.py

```diff
@@ -216,22 +218,37 @@ def load_data_and_make_response(sensor_data_description: dict) -> dict:
         return response
 
     @staticmethod
-    def get_status(sensor: Sensor):
-        end, _ = get_most_recent_clocktime_window(
+    def get_staleness(staleness_search: BeliefsSearchConfigSchema):
+        """Get the staleness of the sensor"""
+        sensor = staleness_search.pop("sensor")
+        _, end = get_most_recent_clocktime_window(
             int(sensor.event_resolution.total_seconds() / 60)
         )
-        test_end = end + sensor.knowledge_horizon(end)
-        start = test_end - (sensor.event_resolution * 4)
 
-        bdf = sensor.search_beliefs(
-            event_starts_after=start,
-            event_ends_before=test_end,
+        bdf1 = sensor.search_beliefs(
```

Comment:
There is a `most_recent_events_only` argument that may be useful here. I think you could then leave out the `event_starts_after` and `event_ends_before` arguments, as to not restrict the time window.